event-handler/teleport_event.go

```diff
@@ -71,6 +83,7 @@ func NewTeleportEvent(e events.AuditEvent, cursor string) (TeleportEvent, error)
 
 		hash := sha256.Sum256(data)
 		id = hex.EncodeToString(hash[:])
+		e.SetID(id)
```

Comment:
If the struct underneath `e` isn't a pointer, then `SetID` wouldn't apply to `i`. Since this section of code can't be certain that `e` is a pointer, it would be safer to move the `var i interface{} = e` line below this line.